packages/tools/src/stateManagement/segmentation/addSegmentationRepresentationsToViewport.ts

```diff
@@ -0,0 +1,153 @@
+import { SegmentationRepresentations } from '../../enums';
+import type { RepresentationPublicInput } from '../../types/SegmentationStateTypes';
+import { internalAddSegmentationRepresentation } from './internalAddSegmentationRepresentation';
+
+/**
+ * Adds one or more segmentations to a specific viewport.
+ */
+export async function addSegmentationRepresentations(
+  viewportId: string,
+  segmentationInputArray: RepresentationPublicInput[]
+) {
+  const promises = segmentationInputArray.map((segmentationInput) => {
+    return internalAddSegmentationRepresentation(viewportId, segmentationInput);
+  });
+
+  Promise.all(promises);
```

Comment:
This isn't an async function as written - just remove the async, and change the signature to:
export function addSegmentationRepresentations(...): Promise<any>

then on line 16, return Promise.all(promise)

You only need the async indicator when you use await within the function body itself.  Otherwise the function is simpler to understand when transpiled just returning a direct promise.

Reply:
Even for a function without await, the async changes the return value and body when it transpiles, so please avoid the keyword unless you actually need it.